src/v2/Apps/Order/Components/CreditCardInput.tsx

```diff
@@ -1,90 +1,103 @@
-import { BorderBox, Sans, color, themeProps } from "@artsy/palette"
-import { fontFamily } from "@artsy/palette/dist/platform/fonts"
-import {
-  BorderProps as InputBorderProps,
-  borderMixin as inputBorder,
-} from "v2/Components/Mixins"
+import { BorderBox, themeProps, useThemeConfig, Text } from "@artsy/palette"
+import { borderMixin, v3BorderMixin } from "v2/Components/Mixins"
 import React from "react"
 import type {
   StripeCardNumberElementChangeEvent,
   StripeError,
 } from "@stripe/stripe-js"
 import { CardElement } from "@stripe/react-stripe-js"
 import styled from "styled-components"
+import { useState } from "react"
+import { fontFamily } from "@artsy/palette/dist/platform/fonts"
 
-export const StyledCardElement = styled(CardElement)`
+export const StyledCardElement = styled(CardElement)<{ isV3?: boolean }>`
   width: 100%;
-  padding: 9px 10px;
+  ${props => (props.isV3 ? "padding: 12px 10px" : "padding: 9px 10px")}
 `
 
 // Re-uses old input border behavior
-const StyledBorderBox = styled(BorderBox)<InputBorderProps>`
-  ${inputBorder};
+export interface BorderProps {
+  hasError?: boolean
+  isV3?: boolean
+}
+
+const StyledBorderBox = styled(BorderBox)<BorderProps>`
   padding: 0;
   height: 40px;
+  ${props => (props.isV3 ? v3BorderMixin : borderMixin)}
+  ${props => props.isV3 && "margin-bottom: 20px"}
 `
 
 interface CreditCardInputProps {
   error?: StripeError
   onChange?: (response: StripeCardNumberElementChangeEvent) => void
 }
 
-interface CreditCardInputState {
-  focused: boolean
-}
+export const CreditCardInput: React.FC<CreditCardInputProps> = props => {
+  const [focused, setFocused] = useState(false)
 
-export class CreditCardInput extends React.Component<
-  CreditCardInputProps,
-  CreditCardInputState
-> {
-  state = {
-    focused: false,
-  }
+  const styles = useThemeConfig({
+    v2: {
+      fontSize: `${themeProps.typeSizes.serif["3t"].fontSize}px`,
+      borderBox: "title",
+      fieldHeight: "40px",
+      fontColor: "black30",
+      isV3: false,
+      fontFamily: fontFamily.serif.regular as string,
+      lineHeight: "20px",
+    },
+    v3: {
+      fontSize: "16px",
+      borderBox: "title",
+      fieldHeight: "50px !important",
+      fontColor: "black60",
+      isV3: true,
+      fontFamily: "inherit",
+      lineHeight: "24px",
```

Comment:
Rather than hardcode these you could import the static themes for each version: https://github.com/artsy/palette/blob/master/packages/palette/src/Theme.tsx#L9 then pull out the tokens.